src/sentry/issues/auto_source_code_config/task.py

```diff
@@ -187,39 +187,22 @@ def create_repos_and_code_mappings(
                 .order_by("-date_added")
                 .first()
             )
-
             if not repository:
+                created = False
                 if not dry_run:
-                    repository = Repository.objects.create(
+                    repository, created = Repository.objects.get_or_create(
                         name=code_mapping.repo.name,
                         organization_id=organization_id,
                         integration_id=organization_integration.integration_id,
                     )
-                metrics.incr(
-                    key=f"{METRIC_PREFIX}.repository.created",
-                    tags={"platform": platform, "dry_run": dry_run},
-                    sample_rate=1.0,
-                )
-
-            extra = {
-                "project_id": project.id,
-                "stack_root": code_mapping.stacktrace_root,
-                "repository_name": code_mapping.repo.name,
-            }
-            # The project and stack_root are unique together
-            existing_code_mappings = RepositoryProjectPathConfig.objects.filter(
-                project=project, stack_root=code_mapping.stacktrace_root
-            )
-            if existing_code_mappings.exists():
-                logger.warning("Investigate.", extra=extra)
```

Comment:
Pass by change.

This has not been happening in production, thus, removing it.